guidance/models/_model_state.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from typing import Optional, overload
+import base64
+import html
+
+
+@dataclass(frozen=True, slots=True)
+class Object:
```

Comment:
We might want a better name....

Reply:
Definitely -- please consider this a sketch 😆